src/main/java/bio/terra/testrunner/runner/UserJourneyResult.java

```diff
@@ -12,13 +14,34 @@
 
   public boolean completed;
   public long elapsedTimeNS;
-  public Exception exceptionThrown;
+
+  public boolean exceptionWasThrown;
+  public String exceptionStackTrace;
+  public String exceptionMessage;
 
   public UserJourneyResult(String userJourneyDescription, String threadName) {
     this.userJourneyDescription = userJourneyDescription;
     this.threadName = threadName;
 
-    this.exceptionThrown = null;
+    this.exceptionWasThrown = false;
+    this.exceptionStackTrace = null;
+    this.exceptionMessage = null;
     this.completed = false;
   }
+
+  /**
+   * Store the exception message and stack trace for the test results. Don't store the full {@link
+   * Throwable} object, because that may not always be serializable. This class is serialized to
+   * disk as part of writing out the test results, so it needs to be a POJO.
+   */
+  public void saveExceptionThrown(Throwable exceptionThrown) {
+    exceptionWasThrown = true;
+    exceptionMessage = exceptionThrown.getMessage();
+
+    StringWriter stackTraceStr = new StringWriter();
+    exceptionThrown.printStackTrace(new PrintWriter(stackTraceStr));
```

Comment:
I wonder if we can make a stylesheet for the JSON output that formats the stack trace attractively. https://www.xml.com/articles/2017/02/14/why-you-should-be-using-xslt-30/

Reply:
I think making some kind of utility to view the test results more easily is a good idea. Even just a script that uses some `jq` incantation to print out any stack traces with newlines would be an improvement.

Added [PF-863](https://broadworkbench.atlassian.net/browse/PF-863) to track this. Please feel free to add suggestions/comments to that also.